serenity-core/src/main/java/net/serenitybdd/core/photography/WebDriverPhotoLens.java

```diff
@@ -43,17 +43,11 @@ private boolean alertIsDisplayedFor(WebDriver driver) {
         if (IsMobile.driver(driver)) { return false; } // No alerts for mobile devices
         if (driver.switchTo() == null) { return false; } // alerts not supported by the driver
 
-        String currentWindow = null;
-        try {
-            currentWindow = driver.getWindowHandle();
-            driver.switchTo().alert().getText();
-            return true;
-        } catch (NoAlertPresentException | UnsupportedCommandException screenshotsNotSupportedIfAnAlertIsPresent) {
+        try{
+            return new WebDriverWait(driver, Duration.ofSeconds(5)).until(ExpectedConditions.alertIsPresent()) != null;
```

Comment:
Add a timeout in the `ThucydidesSystemProperty` class and retrieve the value using the `EnvironmentSpecificConfiguration` class, e.g. 

```
int timeoutInSeconds = EnvironmentSpecificConfiguration.from(environmentVariables)
                    .getOptionalProperty(ThucydidesSystemProperty.WEBDRIVER.WAIT_FOR_ALERT_TIMEOUT)
                    .orElse(0);
```